ci-operator/step-registry/ibmcloud/provision/dedicatedhost/ibmcloud-provision-dedicatedhost-commands.sh

```diff
@@ -0,0 +1,205 @@
+    status=$(${IBMCLOUD_CLI} is dh $dh --output JSON | jq -r ."lifecycle_state")
+
+    run_command "${IBMCLOUD_CLI} is dhd ${dh} -f"
+    run_command "${IBMCLOUD_CLI} is dhgd ${dhg} -f"
+}
+
+function getVMProfile() {
+    local profileHeader=$1
+    local all_profiles str1 str2
+    all_profiles=$(${IBMCLOUD_CLI} is instance-profiles -q | awk '(NR>1) {print $1}')
+    str1=${all_profiles##*$profileHeader}
+    str2=$(echo ${str1}|cut -d " " -f1)
+    echo ${profileHeader}${str2}
+}
+
+#####################################
+##############Initialize#############
+#####################################
+ibmcloud_login
+
+CLUSTER_NAME="${NAMESPACE}-${UNIQUE_HASH}"
+
+resource_group="${CLUSTER_NAME}-rg"
+create_resource_group ${resource_group}
+echo "${resource_group}" > "${SHARED_DIR}/ibmcloud_resource_group"
+
+#####################################
+#######create the dedicated host#####
+#####################################
+
+## Create the dedicated host
+echo "$(date -u --rfc-3339=seconds) - Creating the Dedicated Host..."
+
+dhName="${CLUSTER_NAME}-dh"
+dhProfile1=$(randomGetDHProfile)
+echo "random selected DHProfile1: $dhProfile1"
+
+dhGroup=$(randomSelectRG)
+echo "random select ResourceGroup of Dedicated Host [${dhGroup}]"
+
+zones=("${region}-1" "${region}-2" "${region}-3")
+for zone1 in "${zones[@]}"; do
+    createDH ${dhGroup} ${dhProfile1} ${zone1} ${dhName}
+    if [ -f "${SHARED_DIR}/dedicated_host" ]; then
+        zone=${zone1}
+        break
+    fi
+done
+
+class1=${dhProfile1%%-*}
+
+dhProfile2=$(randomGetDHProfile)
+echo "random selected DHProfile2: $dhProfile2"
+class2=${dhProfile2%%-*}
+
+#Know Issue OCPBUGS-5906 [IPI-IBMCloud] the worker nodes are “Provisioning” when using the dedicated host in the Default group
+if [[ ${dhGroup} == "Default" ]]; then
+    existedDHasWorker=0
+else 
+    existedDHasWorker=$((RANDOM % 2 ))
+fi
+echo "existedDHasWorker: $existedDHasWorker"
+
+if [[ ${existedDHasWorker} == 1 ]]; then
+    echo "create worker nodes in the existed ${dhName}!!"
+    workerProfile=$(getVMProfile "${class1}-4x")
+    masterProfile=$(getVMProfile "${class2}-8x")
+    masterDH="profile: ${dhProfile2}"
+    workerDH="name: ${dhName}"
+else
+    echo "create master nodes in the existed ${dhName}!!"
+    masterProfile=$(getVMProfile "${class1}-8x")
+    workerProfile=$(getVMProfile "${class2}-4x")
+    workerDH="profile: ${dhProfile2}"
+    masterDH="name: ${dhName}"
+fi
+
+#####################################
+##Create dedicated host yaml file ###
+#####################################
+cat > "${SHARED_DIR}/dedicated_host.yaml" << EOF
+controlPlane:
+  platform:
+    ibmcloud:
+      type: ${masterProfile}
+      zones: ["${zone}"]
+      dedicatedHosts:
+      - ${masterDH}
+compute:
+- platform: 
+    ibmcloud:
+      type: ${workerProfile}
+      zones: ["${zone}"]
+      dedicatedHosts:
+      - ${workerDH}
+  replicas: 2
+platform:
+  ibmcloud:
+    resourceGroupName: ${resource_group}
```

Comment:
When using DH, we have to install the cluster in a pre-existing resource group?

Reply:
this is not a must, I am checking whether the dhg and the cluster can be in different resource groups.
I will remove the resource_group in install-config.yaml 
https://mastern-jenkins-csb-openshift-qe.apps.ocp-c1.prod.psi.redhat.com/job/ocp-common/job/Flexy-install/233211/console